File: Molinos.Scato.Dominio/Entidades/Administracion/TarifaPorEmbarque.cs
using Molinos.Scato.Dominio.Dto;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Molinos.Scato.Dominio.Entidades
{
    public class TarifaPorEmbarque : IIdentificable
    {
        [Key]
        public virtual int Id { get; set; }
        public virtual Embarque Embarque { get; set; }
        public virtual Exportador Exportador { get; set; }
        public virtual MaterialPuerto MaterialPuerto { get; set; }
        public virtual DateTime Periodo { get; set; }
        public virtual ICollection<TarifaPorEmbarqueConcepto> TarifaPorEmbarqueConcepto { get; set; }
        public virtual TipoContratoTarifa TipoContratoTarifa { get; set; }
        public virtual bool Cerrado { get; set; }
    }
}

File: Molinos.Scato.WebPuerto/src/app/shared/modelos/administracion/tarifa-por-embarque.ts
import { MaterialPuerto } from "@ScatoModels/material-puerto";
import { Concepto } from "./concepto";
import { Exportador } from "@ScatoModels/exportador";
import { Embarque } from "@ScatoModels/embarque";
import { TipoContratoTarifa } from "./tipo-contrato-tarifa";

export class TarifaPorEmbarque {
    id: number;
    embarque: Embarque;
    materialPuerto: MaterialPuerto;
    exportador: Exportador;
    periodo: Date;
    tarifaPorEmbarqueConcepto: TarifaPorEmbarqueConcepto[];
    tipoContratoTarifa: TipoContratoTarifa;
    cerrado: boolean;
}

export class TarifaPorEmbarqueConcepto {
    id: number;
    tarifaPorEmbarque: TarifaPorEmbarque;
    concepto: Concepto;
    valor: number;
    seleccionado: boolean;
}
